include/checkversion.php

```diff
@@ -25,7 +25,7 @@ function checkversion_run () {
 	if ($checkurl == 'none' ) {
 		return;
 	}
-	$gitversion = dbesc(trim(fetch_url($checkurl)));
+	$gitversion = dbesc(trim(fetch_url("https://raw.githubusercontent.com/friendica/friendica/".$checkurl."/VERSION")));
```

Comment:
This is better, but you still are trusting the user will provide exactly the right part of the version URL. A betterer version would be the following:
````php
switch($checkurl) {
  case 'master': $checked_url = 'https://raw.githubusercontent.com/friendica/friendica/master/VERSION'; break;
  case 'develop': $checked_url = 'https://raw.githubusercontent.com/friendica/friendica/develop/VERSION'; break;
  default: $checked_url = false;
}
````
The default part is important because, again, someone could set the config value to something that isn't `master` or `develop`, but that isn't `none` either, and you want to be able to handle those cases as well.

In general, checking for the exact data you're expecting and discarding all the rest is the best way of avoiding out of bounds bugs.